sdk/src/main/java/io/dapr/client/DaprClientImpl.java

```diff
@@ -273,53 +273,13 @@ public <T extends AbstractStub<T>> T newGrpcStub(String appId, Function<Channel,
   @Override
   public Mono<Void> waitForSidecar(int timeoutInMilliseconds) {
     String[] pathSegments = new String[] { DaprHttp.API_VERSION, "healthz", "outbound"};
-    int maxRetries = 5;
-
-    Retry retrySpec = Retry
-        .fixedDelay(maxRetries, Duration.ofMillis(500))
-        .doBeforeRetry(retrySignal -> {
-          System.out.println("Retrying component health check...");
-        });
-
-    /*
-    NOTE: (Cassie) Uncomment this once it actually gets implemented:
-    https://github.com/grpc/grpc-java/issues/4359
-
-    int maxChannelStateRetries = 5;
-
-    // Retry logic for checking the channel state
-    Retry channelStateRetrySpec = Retry
-            .fixedDelay(maxChannelStateRetries, Duration.ofMillis(500))
-            .doBeforeRetry(retrySignal -> {
-              System.out.println("Retrying channel state check...");
-            });
-    */
 
     // Do the Dapr Http endpoint check to have parity with Dotnet
     Mono<DaprHttp.Response> responseMono = this.httpClient.invokeApi(DaprHttp.HttpMethods.GET.name(), pathSegments,
         null, "", null, null);
 
     return responseMono
-        .retryWhen(retrySpec)
-        /*
-        NOTE: (Cassie) Uncomment this once it actually gets implemented:
-        https://github.com/grpc/grpc-java/issues/4359
-        .flatMap(response -> {
-          // Check the status code
-          int statusCode = response.getStatusCode();
-
-          // Check if the channel's state is READY
-          return Mono.defer(() -> {
-            if (this.channel.getState(true) == ConnectivityState.READY) {
-              // Return true if the status code is in the 2xx range
-              if (statusCode >= 200 && statusCode < 300) {
-                return Mono.empty(); // Continue with the flow
-              }
-            }
-            return Mono.error(new RuntimeException("Health check failed"));
-          }).retryWhen(channelStateRetrySpec);
-        })
-        */
+        .retryWhen(Retry.indefinitely())
```

Comment:
This is great.

However, I would prefer if we left the fixed duration wait between attempts and the log line.

```
 Retry retrySpec = Retry
        .fixedDelay(maxRetries, Duration.ofMillis(500))
        .doBeforeRetry(retrySignal -> {
          System.out.println("Retrying component health check...");
        });
```

This log line is helpful to us because it gives me as an Operator some logging insight into how long its taking the sidecar to become responsive. Also, having no `fixedDelay` between attempts could generate lots of unnecessary log spam which wouldn't be great.

It would also maintain parity with the dotnet SDK

```
public override async Task WaitForSidecarAsync(CancellationToken cancellationToken = default)
        {
            while (true)
            {
                var response = await CheckOutboundHealthAsync(cancellationToken);
                if (response)
                {
                    break;
                }
                await Task.Delay(TimeSpan.FromMilliseconds(500), cancellationToken);
            }
        }
```

Reply:
No problem. I can make that happen. Regarding the log line, println is not the best way to log, so I will change to SLF4J.

Reply:
Agree with SLF4J. It was kinda painful not having any event attributes, such as timestamps/thread id that kinda thing.